### DISCUSSION TITLE
using watch subscription to recalculate values

### DISCUSSION THREAD

**Discussion by alminol** [react-hook-form/react-hook-form#7987] [Q&A]
I am trying to use watch subscription to recalculate some values in base object (eg. when user changes discount percentage or discount amount, the fields are updated accordingly with setValue). The reason why I am doing like this is because I have a lot of recalculating fields and this is just simplifed version. This works roughly something like this:
```
React.useEffect(() => {

    let recalc = true;
    const subscription = watch((data, { name, type }) => {
      if (recalc) {
          recalc = false;
			if (name === "DiscountPercentage") {
				setValue("DiscountAmount", data.GrossAmount * data.DiscountPercentage / 100);	
				setValue("TotalAmount", data.GrossAmount - (data.GrossAmount * data.DiscountPercentage / 100));
			}
			if (name === "DiscountAmount") {
				setValue("DiscountPercentage", data.DiscountAmount / data.GrossAmount * 100);
				setValue("TotalAmount", data.GrossAmount - data.DiscountAmount);
			}
          recalc = true;
        }
      }
    });
    return () => {
      recalc = false;
      subscription.unsubscribe();
    };
  }, [watch, setValue]);
```

This works great, except the fact that setValue triggers watch again from inside recalculation, user changes DiscountPercentage and it triggers DiscountAmount again, which causes problems (eg. sometimes the discount percentage gets changed from 20% to 19.98% as a feedback because it's rounding values to 2 decimals)

Is there any way to block or pause watch subscription when using setValue in callback method like in this particular case?

Thanks...

**Comment by bluebill1049** [comment]
> Is there any way to block or pause watch subscription when using setValue in callback method like in this particular case?

not at this stage.
